crates/ruff_python_ast/src/find_node.rs

```diff
@@ -42,10 +42,17 @@ pub(crate) fn covering_node(root: AnyNodeRef, range: TextRange) -> CoveringNode
         "Range is not contained within root"
     );
 
+    // When we visit the root it won't `enter_node` so do that manually.
+    // Can this check fail? I don't know. It's fine.
+    let ancestors = if root.range().contains_range(range) {
+        vec![root]
+    } else {
+        Vec::new()
+    };
```

Comment:
This change was required to make `ModModule` be included, or else we couldn't find the sibling Statement in a global attribute of a module. This had a drive-by side-effect on selection_range that I'm not sure if it's correct/problematic.

Reply:
How's this different from the check on line 59? Can we remove the check on line 59? 

I think a better solution is to add a `walk_node` method to the `source_order` module like this:

```rust
pub fn walk_node<'a, V>(visitor: &mut V, node: AnyNodeRef<'a>)
where
    V: SourceOrderVisitor<'a> + ?Sized,
{
    if visitor.enter_node(node).is_traverse() {
				node.visit_source_order(visitor);
    }

    visitor.leave_node(node);
}
```


and call that over `node.visit_source_order` and remove the check on line 59